spec/helpers/compliance_summary_helper_spec.rb

```diff
@@ -18,23 +18,25 @@
 
     it "#textual_compliance_status" do
       @record.compliances = [@compliance1]
+      @record.save
       date = @compliance1.timestamp
       expect(helper.textual_compliance_status).to eq(:label    => "Status",
                                                      :icon     => "pficon pficon-ok",
                                                      :value    => "Compliant as of #{time_ago_in_words(date.in_time_zone(Time.zone)).titleize} Ago",
                                                      :title    => "Show Details of Compliance Check on #{format_timezone(date)}",
                                                      :explorer => true,
-                                                     :link     => "/vm_infra/show?count=1&display=compliance_history")
+                                                     :link     => "/vm_infra/show/#{@record.id}?count=1&display=compliance_history")
```

Comment:
I guess I am a bit confused by this line change.  The `@record.save` make sense conceptually to me, but I am unsure why the `:link` would need to change as well.

Reply:
Aah, because the link gets generated using `@record.id` but when the record was never saved, the id was nil

Reply:
Ah, gotcha.  `@record` also wasn't saved prior to this, not just the associated compliance records.  That makes sense now.